ice.go

```diff
@@ -141,3 +160,87 @@ func (c *ICECandidate) Unmarshal(raw string) error {
 
 	return nil
 }
+
+// GetICEParams extracts session-level ICE parameters
+func (s *SessionDescription) GetICEParams() ICESessionParameters {
+	ret := ICESessionParameters{}
+
+	var value string
+	var found bool
+
+	_, found = s.Attribute("ice-lite")
+	if found {
+		ret.Lite = true
+	} else {
+		ret.Lite = false
+	}
+	value, found = s.Attribute("ice-ufrag")
+	if found {
+		ret.UserFragment = value
+	} else {
+		ret.UserFragment = ""
+	}
+	value, found = s.Attribute("ice-pwd")
+	if found {
+		ret.Password = value
+	} else {
+		ret.Password = ""
+	}
+	value, found = s.Attribute("ice-options")
+	if found {
+		ret.Options = value
+	} else {
+		ret.Options = ""
+	}
+
+	return ret
+}
+
+// GetICEParams extracts media-section-level ICE parameters
+func (s *MediaDescription) GetICEParams() ICEMediaParameters {
+	ret := ICEMediaParameters{}
+
+	var value string
```

Comment:
I would put a little interface in this file that is un-exported then you don't have to duplicate `(s *SessionDescription) GetICEParams()`